File: cc/021.cc
// Project Euler Problem 021
//
// Evaluate the sum of all amicable pairs under 10000.

#include <cstdio>
#include <vector>
#include "primes1.h"

using namespace std;

int main(int argc, const char* argv[]) {
  Primes1<int> primes;
  int sum = 0;
  for (int i=2; i < 10000; i++) {
    int sd = primes.sum_divisors(i);
    if ((sd > i) && (primes.sum_divisors(sd)==i)) {
      sum += i + sd;
    }
  }
  printf("%d\n", sum);
}

File: go/021.go
// Project Euler Problem 021
//
// Evaluate the sum of all amicable pairs under 10000.

package main

import (
	"fmt"

	"./primes"
)

func main() {
	sum := int64(0)
	for i := int64(2); i < 10000; i++ {
		sd := primes.SumDivisors(i)
		if sd > i && primes.SumDivisors(sd) == i {
			sum += i + sd
		}
	}
	fmt.Println(sum)
}
